rfcs/0000-cosign-integration.md

```diff
@@ -0,0 +1,167 @@
+# Integrate `cosign` with `kpack`
+
+## Problem
+
+The signing solution currently supported by `kpack` is
+[Notary v1](https://github.com/theupdateframework/notary). However, Notary
+presents itself with several different challenges, including:
+- It is unable to sign container image digests. The user must always specify a
+tag to be signed.
+- It is unable to verify container image signature by digests.
+- Signature relocation is not possible with this tool.
+- It is a project that currently has low activity and community involvement,
+  since the community is currently focused on the development of Notary v2.
+
+[`cosign`](https://github.com/sigstore/cosign) presents itself as a solution to
+some of those aforementioned challenges. With `cosign`, the user is able to:
+- Relocate container images across registries along with its signatures, using
+  either:
+  - `crane cp` to copy images and signatures separately.
+  - `cosign copy` to copy images along with their signatures from one registry
+    to another.
+- Sign container images by tag or by digest.
+- Verify container image signatures by specifying either a tag or a digest.
+- Verify container image signatures after relocation.
+
+Moreover, `cosign` is a project that has had a lot of activity and community
+involvement recently.
+
+## Outcome
+
+`kpack` integrates with `cosign` to sign images it builds and push the
+signatures to a registry so that users can ensure the chain of custody of a
+generated artifact.
+
+This proposal aims to cover specifically the flow of signing an image produced
+by the `kpack Image` resource, along with its builder image, without
```

Comment:
```suggestion
by the kpack `Image`, `Build`, `Builder` and `ClusterBuilder` resources, without
```
Because the the Builder resources are unique resources that may not be used with the Image resource.

Reply:
@matthewmcnew Makes sense! Is it OK if I keep the preformatting for the `kpack` word?